### PR TITLE
fix: don't generate coverage data for deps/

### PR DESCRIPTION
**Pull request by bklebe**
#### Summary of changes
**Asana Ticket:** [fix Arrow coverage reporting](https://app.asana.com/0/584764604969369/1201120775201216/f)

Tells Lcov to ignore the `deps/` path when generating coverage info. We don't want `deps/mox/lib/mox.ex` covered and it breaks coverage reporting.

#### Reviewer Checklist
- [ ] Meets ticket's acceptance criteria
- [ ] Any new or changed functions have typespecs
- [ ] Tests were added for any new functionality (don't just rely on Codecov)
- [ ] This branch was deployed to the staging environment and is currently running with no unexpected increase in warnings, and no errors or crashes.

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -22,7 +22,7 @@ defmodule Arrow.MixProject do
         ignore_warnings: ".dialyzer.ignore-warnings"
       ],
       preferred_cli_env: ["test.integration": :test],
-      test_coverage: [tool: LcovEx]
+      test_coverage: [tool: LcovEx, ignore_paths: ["deps/"]]
```
Comment (paulswartz) [inline mix.exs:25]: Is this something that we should include in the GitHub Actions migration doc?

**Review by paulswartz — APPROVED** [review]
🍰

File: mix.exs (hunk 1 above)
Comment (bklebe) [reply mix.exs:25]: Probably? I think `lcov.info` will contain coverage info for deps that are used in tests if not. `realtime_signs` uses Mox and doesn't seem to have this problem, but `api` does. Not sure what the difference is.

Comment (bklebe) [reply mix.exs:25]: Actually `api` seems to have a different problem. Disregard.

**Pull request merged**